docs/downloader.rst

```diff
@@ -0,0 +1,169 @@
+
+Each configured magic-folder has a ``collective_dircap`` which is a Tahoe-LAFS Directory Capability ("dircap") for the list of participants.
+If this dircap is writable then this device is the administrator (and the only device which can modify the participants).
+
+In either case, the capability can be downloaded.
+It will be a Tahoe-LAFS directory containing a series of sub-directories; these are the participants.
+The directory name is the Participant name and points at a directory read-capability where all the files in their magic-folder are stored.
+This directory is known as the "Personal DMD" (where DMD stands for "Distributed Mutable Directory").
+
+The entries in a user's Personal DMD are flat (no subdirectories) and point a (mangled) relative path-name to a Snapshot.
+
+So, Snapshots to download are discovered by:
+
+- reading the Collective DMD
+- for each user in it:
+  - read their Personal DMD
+  - for each entry in the Personal DMD:
+    - queue the Snapshot for caching, including all parents
+    - after download, queue the Snapshot for updates
+
+Note that we don't always download *every* parent; we can stop early if a common ancestor is found.
+
+
+Downloading Snapshots
+---------------------
+
+The ``RemoteSnapshotCacheService`` awaits capability-strings of Snapshots to download.
+For each one, the function ``create_snapshot_from_capability`` is used to download the capability and return a ``RemoteSnapshot`` instance.
+This ``RemoteSnapshot`` is stored in memory.
+
+Additionally, all parents of the Snapshot are also cached.
+We can stop caching parents once we find a "common ancestor"; this means a parent in the remote snapshot that matches the one in our Personal DMD.
+
+Conflict Resolution is described in :ref:`Multi-party Conflict Detection` under the Leif's Design section.
+Briefly: a ``RemoteSnapshot`` is traced through its parents until a common ancestor is found.
+If the new Snapshot is a descendant of our latest Snapshot for that name, it's an overwrite.
+If it is not, there is a conflict (unless we don't yet have that name at all, then it's a creation).
+
+
+On Overwrite
+------------
+
+The ``content`` of the ``RemoteSnapshot`` is downloaded and moved into place in our Magic Folder.
+Our Personal DMD is updated to point at this Snapshot.
+
+In case there is no ``content`` this is a delete and we simply remove the corresponding local file.
+
+Note that a completely new file (a "create") is the same as a normal overwrite (except of course there's no possibility of a conflict).
+
+
+On Conflict
+-----------
+
+The ``content`` of the ``RemoteSnapshot`` is downloaded and moved into a "conflict file" (see Leif Design) beside the conflicting content.
+The Personal DMD is **not** updated.
+Once the conflict is "resolved" then a new Snapshot is created with two parents: the latest
+Snapshot we had at conflict time and the conflicting Snapshot.
+Our Personal DMD is updated to point at this new Snapshot.
+
```

Comment:
Given that this is *not* implemented, we should describe what we currently do when we encounter a conflict.

Reply:
Adding another `(Not implemented)` to be more clear that we don't (yet) support resolving conflicts.